code/controllers/subsystem/garbage.dm

```diff
@@ -279,6 +279,7 @@ SUBSYSTEM_DEF(garbage)
 				SSgarbage.Queue(D)
 			if (QDEL_HINT_IWILLGC)
 				D.gc_destroyed = world.time
+				SSdemo.mark_destroyed(D)
```

Comment:
Can you explain why it's only in a few specific spots here?

Reply:
I need to research more about internal workings of this beyond a rough idea of how it works to give you a solid answer on this. Maybe @monster860 would know directly.

Reply:
Seems like it's only on the ones that fully destroy and aren't for testing